sdks/go/pkg/beam/runners/prism/internal/jobservices/management.go

```diff
@@ -182,8 +182,6 @@ func (s *Server) Prepare(ctx context.Context, req *jobpb.PrepareJobRequest) (_ *
 				check("TimerFamilySpecs.TimeDomain.Urn", spec.GetTimeDomain(), pipepb.TimeDomain_EVENT_TIME, pipepb.TimeDomain_PROCESSING_TIME)
 			}
 
-			check("OnWindowExpirationTimerFamily", pardo.GetOnWindowExpirationTimerFamilySpec(), "") // Unsupported for now.
```

Comment:
I ask this here as it's where the Pipeline graph is first analyzed. Is it the responsibility of the SDK or the Runner to check whether the input PCollection is a PCollection<KV>?

Reply:
I don't quite understand this question. What would an input PCollection be if it wasn't a PCollection?

In some ways it's both.

I can say it's the SDK's responsibility to send the runner well formed Directed Acyclic Graphs as Pipelines. This lets it produce better error messages than the runner can.
But, it's the Runner's responsibility to validate that the SDK is doing so, and report that as an error or not.